### ISSUE TITLE
Package Request: Pandoc

### ISSUE THREAD

**Issue by dominicschaff** [termux/termux-packages#372]
I use [pandoc](http://pandoc.org/) on all my machines and I would really like to have it on termux, then I can use my device for almost all my computer usage.

**Comment by PurpleGuitar** [comment]
I second this request. Having Pandoc available on my Android devices would be a big help.

**Comment by Mihara** [comment]
I spent some time researching that, that is, trying to muck through building a binary. Any binary. No success so far.

The sum total of my findings is as follows:

* There is only one GHC build kit set up for cross-compilation: docker `sseefried/debian-wheezy-ghc-android` which is apparently based on https://github.com/neurocyte/ghc-android/ The docker image is a bit out of date, and, building it from scratch fails, but the image works for suitably small quantities of 'works'.
* It took me bloody hours to figure out how to invoke the cross-compiler. `arm-linux-androideabi-cabal` is apparently the way to go.
* Specific pitfalls: 
    * hslua trips over an android library not telling it what a decimal point is. Solution described at https://stackoverflow.com/questions/12299659/compiling-lua-lib-for-android-success-but-strange-segfaults -- hslua needs to be unpacked and built patched. The exact bug is different, but the reason is the same.
    * zlib can't be crosscompiled. patch: http://lpaste.net/8287391336118616064 works for version 0.5.4.2 but not for later versions.
    * network-2.6.3.1 can't be crosscompiled. 2.5.0.0 apparently can.

The final stumbling point was the JuicyPixels library which chokes on this:
```
Loading package ghc-prim ... linking ... ghc: /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/lib/arm-unknown-linux-androideabi-ghc-7.8.3/ghc-prim-0.3.1.0/HSghc-prim-0.3.1.0.o: unhandled ELF relocation(Rel) type 28

ghc: unable to load package `ghc-prim'
```

I googled about for a while, but found no solution -- see https://github.com/neurocyte/ghc-android/issues/25

So far, it looks kinda grim, and since I don't know my way around GHC or Haskell at all, I don't have any more ideas.

**Comment by brownstyler** [comment]
I third this request, I use emacs and org mode on termux quite a bit and having pandoc would be amazing!

**Comment by bjornekstrom** [comment]
Indeed, pandoc would be a great way for me to make pdfs from Markdown in Termux.

**Comment by ShadowJonathan** [comment]
I was writing some stories on my phone, and wanted to convert it (markdown) to HTML for a friend to view, on the go, so i second this, pandoc is great.

**Comment by sdrausty** [comment]
Haskel-pandoc is available https://www.archlinux.org/packages/community/x86_64/pandoc/ for x86_64 devices in Arch Linux in Termux PRoot.  If you have success with pandoc in Termux on Android, Chromebook and Fire OS, post here.

**Comment by borghal** [comment]
I'd love to be able to use pandoc, too. Has anybody successfully installed it?

**Comment by sw2000** [comment]
Probably not the most efficient way, but if it helps anyone, I've got R, pandoc and pandoc-citeproc running fine with Termux on arm Android after installing Debian using [debian_on_termux.sh](https://github.com/sp4rkie/debian-on-termux). You need to get pandoc v2.2 from the debian buster testing repository, as the default installation (1.12) has segmentation fault issues. Buster also has lots of R libraries precompiled such as r-cran-tidyverse, which avoids tedious compiling issues on low RAM devices.

**Comment by jlrn7** [comment]
> For anyone who's still looking for a solution for arm devices, _(I did not try the Arch way cited above and don't know if it will or will not work on arm. For x86_64 it's not really worth it, since a static binary is available and should work.)_ pandoc is available in https://github.com/xeffyr/termux-extra-packages
> 

I downloaded the unstable pandoc package, and it runs miserably. Sending info in the image attached. Apparently this is an issue with the glibc library on this particular device.

![Screenshot_20190809-202303 1](https://user-images.githubusercontent.com/12088563/62815836-222a0600-bae4-11e9-91ee-51bab3860337.png)

On the other hand, I could compile and install other packages seamlessly.

**Comment by xeffyr** [comment]
> Apparently this is an issue with the glibc library on this particular device.

It's with **QEMU** issue and **glib** (not glibc !) shows errors.

**Comment by JosiasAurel** [comment]
Has anyone found a way to have pandoc on termux ?

**Comment by alerque** [comment]
Fedora Rawhide has Pandoc packages compiled for `armv7hl`. What target architecture is needed on a typical modern Android phone?

**Comment by xeffyr** [comment]
> What target architecture is needed on a typical modern Android phone?

GHC toolchain compatible with Android OS (NDK, Bionic libc). Unfortunately, none of maintainers are familiar with Haskell GHC and build tools.

Package request labels well describe what we need:
![Снимок экрана от 2020-10-15 18-23-35](https://user-images.githubusercontent.com/25881154/96151174-cba02f00-0f13-11eb-947e-e8eba685f35e.png)

**Comment by hltdev8642** [comment]
to anyone who wants a working pandoc aarch64 static binary, I found this repo and so far it seems to work without any hiccups!

I guess it was built for a raspberry pi,  but it apparently works all the same on Android in termux. 

[repo](https://github.com/arm4rpi/pandoc-arm)
[binary release that worked for me](https://github.com/arm4rpi/pandoc-arm/releases/download/v0.1/pandoc-2.9.2-aarch64-static.tar.gz)

**Comment by iT-Boyer** [comment]
> to anyone who wants a working pandoc aarch64 static binary, I found this repo and so far it seems to work without any hiccups!
> 
> I guess it was built for a raspberry pi,  but it apparently works all the same on Android in termux. 
> 
> [repo](https://github.com/arm4rpi/pandoc-arm)
> [binary release that worked for me](https://github.com/arm4rpi/pandoc-arm/releases/download/v0.1/pandoc-2.9.2-aarch64-static.tar.gz)
> 
> 
How to install the package for pandoc?

**Comment by hltdev8642** [comment]
just extract it to a directory in your $PATH, (or just run it locally by using `./pandoc` or `bash ./pandoc`.

Otherwise copy it into ~/bin , /usr/bin , or somewhere else in your path :-]

**Comment by iT-Boyer** [comment]
Thanks, it can work.

**Comment by RFDAJE** [comment]
another solution to install pandoc
- install https://github.com/t184256/nix-on-droid per f-droid
- `nix-env --install pandoc`

**Comment by milahu450** [comment]
> installing Debian using [debian_on_termux.sh](https://github.com/sp4rkie/debian-on-termux)

```sh
wget -q https://raw.githubusercontent.com/sp4rkie/debian-on-termux/master/debian_on_termux_10.sh &&
sh debian_on_termux_10.sh

apt install pandoc
```

ok. now how do i share files between termux and debian?

**Comment by milahu450** [comment]
bonus points for wrapping the debian env
so we can call pandoc from termux

**related**

https://github.com/milahu450/alchi/blob/62c900e6310da9539ad6d3b2425ad3bd2a94abd6/src/scripts/pandoc-on-termux.sh

https://github.com/milahu/nur-packages/tree/master/pkgs/applications/audio/qaac
